### PR TITLE
fix(plugins-view): Register custom plugin registries from devfile plugins

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Scan devfile plugins and if it has custom plugin registry source, add it's plugin registry to che-theia's plugin registry cache.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19898

### How to test this PR?

1. Start a workspace from this devfile:
```
apiVersion: 1.0.0
metadata:
  name: wksp-uhias
components:
  - type: cheEditor
    reference: 'https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1131/che_theia_meta.yaml'
    alias: che-theia
  - id: editorconfig/editorconfig/0.14.4
    type: chePlugin
    registryUrl: 'http://vinokur-registry.surge.sh/path/to/registry/v3/'
```
2. Open the Plugins view `View => Plugins`.
3. Click the three dots button and select `Show installed plugins`.
See: the plugin `editorconfig` is shown in the list, the icon is displayed as well.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Review by ericwill — APPROVED** [review]

File: extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts [hunk 1]
```diff
@@ -222,6 +226,17 @@ export class ChePluginManager {
       registries[registry.name] = registry;
     }
 
+    const devfile = await this.devfileService.get();
+    if (devfile.components) {
+      const components = devfile.components.filter(component => component.plugin && component.plugin.id && component.plugin.registryUrl);
+      for (let i = 0; i < components.length; i++) {
+        const component = components[i];
```
Comment (RomanNikitenko) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts:233]: can be simplified like: 
```
for (const component of components) {
```
but up to you

**Comment by RomanNikitenko** [comment]
I tested and can see installed plugin
plugin_registry

I noticed a warning
invsalid_plugin_reg

@vinokurig 
is it related to your changes?

File: extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts (hunk 1 above)
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts]: done

**Comment by RomanNikitenko** [comment]
It looks like I have the same behavior for the `next` version of the editor, so without your changes:
master_plugin_registry

Do you know - is it a bug or I'm doing something wrong?
Should I create a separate issue for that problem?

**Comment by RomanNikitenko** [comment]
Created a separate issue https://github.com/eclipse/che/issues/19955

**Review by RomanNikitenko — APPROVED** [review]

**Review by vitaliy-guliy — APPROVED** [review]

**Pull request merged**